app/celery/process_ses_receipts_tasks.py

```diff
@@ -39,7 +39,7 @@ def process_ses_results(self, response):
         reference = ses_message['mail']['messageId']
 
         try:
-            notification = notifications_dao.dao_get_notification_by_reference(reference)
+            notification = notifications_dao.dao_get_notification_history_by_reference(reference=reference)
```

Comment:
optional: is it possible for us to rename this function? took me quite a while to realise that it checks notifications first, the notification_history. i'd prefer something like `dao_get_notification_or_history_by_reference`